src/wiringX.c

```diff
@@ -69,10 +69,148 @@ struct spi_t {
 	int fd;
 } spi_t;
 
-static struct spi_t spi[2] = {
-	{ 0, 0, 0, 0, 0 },
-	{ 0, 0, 0, 0, 0 }
+/*
+ * Linked List for storing state of spi connections
+ * * Efficient for small number of active connections
+ * * Doesn't make any assumptions on device and channel numbers
+ */
+struct ll_spi_t {
+	int device;
+	int channel;
+	struct spi_t data;
+	struct ll_spi_t *next;
 };
+struct ll_spi_t *spi;
+
+/*
+ * look up state for given combination of device and channel number
+ * returns null-pointer on error
+ */
+struct spi_t *spi_get(int device, int channel) {
+	struct ll_spi_t *ptr = spi;
+	while(ptr != NULL && ptr->device != device && ptr->channel != channel)
+		ptr = ptr->next;
+
+	if(ptr != NULL)
+		return &ptr->data;
+
+	return NULL;
+}
+
+/*
+ * creates new state for given device channel combination
+ * clears existing state if any
+ */
+struct spi_t *spi_new(int device, int channel) {
+	struct ll_spi_t *ptr = spi;
+	
+	/* list start */
+	if(ptr == NULL) {
+		ptr = calloc(sizeof(struct ll_spi_t), 1);
```

Comment:
just figured it may be acceptable to just segfault in that case. But I am quite open to implementing a check.